src/Sarif.WorkItems/SarifWorkItemFiler.cs

```diff
@@ -79,65 +87,82 @@ public virtual void FileWorkItems(string sarifLogFileContents)
-            if (optionallyEmittedData != OptionallyEmittedData.None)
+            using (IDisposable loggingScope = Logger.BeginScope("FileWorkItems"))
             {
-                var dataInsertingVisitor = new InsertOptionalDataVisitor(optionallyEmittedData);
-                dataInsertingVisitor.Visit(sarifLog);
-            }
+                this.FilingSucceeded = false;
+                this.FiledWorkItems = new List<WorkItemModel>();
 
-            SplittingStrategy splittingStrategy = this.FilingContext.SplittingStrategy;
-            if (splittingStrategy == SplittingStrategy.None)
-            {
-                FileWorkItemsHelper(sarifLog, this.FilingContext, this.FilingClient);
-                return;
-            }
+                sarifLog = sarifLog ?? throw new ArgumentNullException(nameof(sarifLog));
 
-            IList<SarifLog> logsToProcess = new List<SarifLog>(new SarifLog[] { sarifLog });
+                Logger.LogInformation("Connecting to filing client: {accountOrOrganization}", this.FilingClient.AccountOrOrganization);
+                this.FilingClient.Connect(this.FilingContext.PersonalAccessToken).Wait();
 
-            PartitionFunction<string> partitionFunction = null;
+                OptionallyEmittedData optionallyEmittedData = this.FilingContext.DataToRemove;
+                if (optionallyEmittedData != OptionallyEmittedData.None)
+                {
+                    var dataRemovingVisitor = new RemoveOptionalDataVisitor(optionallyEmittedData);
+                    dataRemovingVisitor.Visit(sarifLog);
+                }
 
-            switch (splittingStrategy)
-            {
-                case SplittingStrategy.PerRun:
+                optionallyEmittedData = this.FilingContext.DataToInsert;
+                if (optionallyEmittedData != OptionallyEmittedData.None)
                 {
-                    partitionFunction = (result) => result.ShouldBeFiled() ? "Include" : null;
-                    break;
+                    var dataInsertingVisitor = new InsertOptionalDataVisitor(optionallyEmittedData);
+                    dataInsertingVisitor.Visit(sarifLog);
                 }
-                case SplittingStrategy.PerResult:
+
+                SplittingStrategy splittingStrategy = this.FilingContext.SplittingStrategy;
+
+                if (splittingStrategy == SplittingStrategy.None)
                 {
-                    partitionFunction = (result) => result.ShouldBeFiled() ? Guid.NewGuid().ToString() : null;
-                    break;
+                    FileWorkItemsHelper(sarifLog, this.FilingContext, this.FilingClient);
+                    return;
                 }
-                default:
+
+                IList<SarifLog> logsToProcess = new List<SarifLog>(new SarifLog[] { sarifLog });
+
+                PartitionFunction<string> partitionFunction = null;
+
+                Stopwatch splittingStopwatch = Stopwatch.StartNew();
+                
```

Comment:
Did you add this just to include in the logging below?

Reply:
Yes. It's really just an example. I doubt we really care about the perf of splitting the SARIF files. I suspect we'll remove this when we add the real logging messages.

---
In reply to: [395896508](https://github.com/microsoft/sarif-sdk/pull/1820#discussion_r395896508) [](ancestors = 395896508)

Reply:
Great. Thanks. I was confused by it at first until I saw you use it and wanted to confirm it was just an example.